src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs

```diff
@@ -457,17 +424,23 @@ public override string ToString()
                 : DotNetFrameworkName;
         }
 
-        public bool Equals(NuGetFramework other)
+        public bool Equals(NuGetFramework? other)
         {
+#pragma warning disable CS8604 // Possible null reference argument.
```

Comment:
We'd have to change `Comparer` to be of type `NuGetFrameworkFullComparer` instead of `IEqualityComparer<NuGetFramework>`. However, that's a public API, so it would be a breaking change.

As the comment on the next line says/implies, .NET Framework and .NET Standard (and .NETCoreApp before version 5) don't have nullable annotations, and the compiler ("incorrectly") thinks this means nulls are not allowed. But the compiler just assumes that methods without nullable annotations don't allow nulls, which isn't correct in this situation.

Reply:
Ah, makes sense!